src/components/PressableWithSecondaryInteraction/index.native.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {forwardRef} from 'react';
+import {GestureResponderEvent, TextProps} from 'react-native';
+import {PressableRef} from '@components/Pressable/GenericPressable/types';
+import PressableWithFeedback from '@components/Pressable/PressableWithFeedback';
+import Text from '@components/Text';
+import PressableWithSecondaryInteractionProps from './types';
+
+/** This is a special Pressable that calls onSecondaryInteraction when LongPressed. */
+function PressableWithSecondaryInteraction(
+    {
+        children,
+        onSecondaryInteraction,
+        inline = false,
+        needsOffscreenAlphaCompositing = false,
+        suppressHighlighting = false,
+        activeOpacity = 1,
+        preventDefaultContextMenu,
+        withoutFocusOnSecondaryInteraction,
+        enableLongPressWithHover,
+        ...rest
+    }: PressableWithSecondaryInteractionProps,
+    ref: PressableRef,
+) {
+    const executeSecondaryInteraction = (event: GestureResponderEvent) => {
+        event.preventDefault();
+        onSecondaryInteraction?.(event);
+    };
+
+    // Use Text node for inline mode to prevent content overflow.
+    if (inline) {
+        return (
+            <Text
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...(rest as TextProps)}
+                suppressHighlighting={suppressHighlighting}
+                onLongPress={onSecondaryInteraction ? executeSecondaryInteraction : undefined}
+            >
+                {children}
+            </Text>
+        );
+    }
+
+    return (
+        <PressableWithFeedback
+            // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
Let's add a comment explaining why we're disabling the lint rule here

Reply:
Done!